commerce/bookstore/python/load_data.py

```diff
@@ -0,0 +1,286 @@
+                                           'insert $th (sub-tag: $b, sup-tag: $p) isa tag-hierarchy;')
+                transaction.query().insert('match $b = "Technical Documentation"; $b isa genre;'
+                                           '$p = "Adults only"; $p isa genre;'
+                                           'insert $th (sub-tag: $b, sup-tag: $p) isa tag-hierarchy;')
+                transaction.query().insert('insert $g "Map" isa genre;')
+                transaction.query().insert('match $b = "Map"; $b isa genre;'
+                                           '$p = "Technical Documentation"; $p isa genre;'
+                                           'insert $th (sub-tag: $b, sup-tag: $p) isa tag-hierarchy;')
+                transaction.commit()
+    print('Loaded genre tags.')
+    print('\nAll data loaded successful!')
+    return
+
+
+def load_data():  # Main data load function
+    with TypeDB.core_client("localhost:1729") as client:
+        with client.session(db, SessionType.DATA) as session:
+            for input in Inputs:
+                input["file"] = data_path + input["file"]
+                print("Loading from [" + input["file"] + ".csv] into TypeDB ...")
+                load_data_into_typedb(input, session)  # Main data loading function. Repeat for only file in Inputs
+            generate_ordered_items()  # Add randomly generated lists of items into orders
+            load_genre_tags()  # Load genre tags
+    return
+
+
+def check_data():  # Checking whether the DB has schema and data already
+    with TypeDB.core_client("localhost:1729") as client:
+        with client.session(db, SessionType.SCHEMA) as session:
+            with session.transaction(TransactionType.READ) as transaction:
+                try:
+                    TypeQL_read_query = 'match $b isa Book, has ISBN $x; get $x; limit 3;'
+                    transaction.query().match(TypeQL_read_query)
+                    print('The DB contains the schema and loaded data already.')
+                    return 1
+                except:  # If the attempt was unsuccessful — we consider DB as empty (brand new, no schema)
+                    return 0
+
+
+def setup():  # Loading schema
+    with TypeDB.core_client("localhost:1729") as client:
+        with client.session(db, SessionType.SCHEMA) as session:
+            with open('../schema.tql', 'r') as schema:
+                define_query = schema.read()
+                with session.transaction(TransactionType.WRITE) as transaction:
+                    try:
+                        transaction.query().define(define_query)
+                        transaction.commit()
+                        print("Loaded the " + db + " schema.")
+                        return 0
+                    except Exception as e:
+                        print('Failed to load schema: ' + str(e))
+                        return 1
+
+
+# This is a list of imported files with datasets for the DB
+Inputs = [
```

Comment:
This feel like something we should type a bit more strongly. (We're quite big on strong typing.) How about introducing an `Input` class?

Reply:
I'm not sure I follow - It is not used anywhere outside of loading the data. It does not need to be in the schema. It's basically just a list of .csv file names and the correlated functions to load them.

Reply:
I have tried to do that in a separate PR here https://github.com/izmalk/typedb-examples/pull/1/files 
But I feel like the result is by far worse than the initial variant. I suggest discussing in this separate PR how to proceed. Maybe it's just a problem with my implementation.

Reply:
I have rebased the PR with classes on the latest version of this repository. So it's ready to be merged

Reply:
Setup the database - by loading the schema.
If you want I can rename it to something like `load_schema`?

Reply:
Fixed in https://github.com/vaticle/typedb-examples/pull/143/commits/06554953e94afbf6c447c02db3032f8d63a18114